src/python/pants/backend/python/target_types_test.py

```diff
@@ -146,6 +146,7 @@ def test_infer_pex_binary_entry_point_dependency(caplog) -> None:
         ],
         target_types=[PexBinary, PythonRequirementTarget, PythonSourcesGeneratorTarget],
     )
+    rule_runner.set_options(["--python-infer-unowned-dependency-behavior=error"])
```

Comment:
Is there any way to add tests instead of just changing the semantics of the existing tests?

Reply:
I tried parametrising the whole test. I'm not sure that it's clearer this way, though. I could split out the region that's different into 3 separate tests so there isn't control flow. Thoughts?

Reply:
I just want to make sure we have good coverage for each of the possible values